components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/factory/HttpRegistrationResponseFactoryTest.java

```diff
@@ -89,6 +94,26 @@ public void testCanHandle(Object identityResponse, boolean expected) throws Exce
         }
     }
 
+    @DataProvider(name = "exceptionInstanceProvider")
+    public Object[][] getExceptionInstanceType() {
+
+        FrameworkException exception1 = new RegistrationException("");
+        FrameworkException exception2 = new FrameworkException("");
+        return new Object[][]{
+                {exception1, true},
+                {exception2, false}
+        };
+    }
+
+    @Test(dataProvider = "exceptionInstanceProvider")
```

Comment:
Can't we use tests with `expectedExceprions` here?

Reply:
here I checked two different exceptions and both related to unhappy path. if used expectedException have to use two separate tests. as both covers unhappy path I used this method.